draft-ietf-cose-dilithium.md

```diff
@@ -89,16 +89,20 @@ Some examples in this specification are truncated using "..." for readability.
 
 This section describes a generic cryptographic key structure for use with algorithms not limited to those registered in this document.
 The Algorithm Key Pair (AKP) Type is used to express Public and Private Keys for use with Algorithms.
-When this key type is used the "alg" JSON Web Key Parameter or COSE Key Common Parameter is REQUIRED.
+The concept of public and private information classes for key pairs originates from {{Section 8.1 of RFC7517}}.
+The parameters for public and private information classes contain byte strings in a format specified by the "alg" value.
+The "alg" JSON Web Key Parameter or COSE Key Common Parameter is REQUIRED for all AKP keys.
+The "pub" parameter contains public information and is REQUIRED.
+The "priv" parameter contains private information and MUST NOT be present in public keys.
 
-The "pub" parameter contains a public key, this parameter contains public information and is REQUIRED.
-The "priv" parameter contains a private key, sometimes called a secret key, this parameter contains private information.
-The concept of public and private information classes originates from {{Section 8.1 of RFC7517}}.
+Some algorithm specifications, such as ML-DSA, might allow for use of multiple key type parameters for private information.
+When registering new algorithms, use of multiple key type parameters for private information is NOT RECOMMENDED.
 
-The "pub" and "priv" parameters contain byte strings in a format specified by the "alg" value.
-The values of "pub" and "priv" MAY have additional structure or length checks depending on the associated "alg" parameter and its requirements.
+The "seed" parameter contains private information and MUST NOT be present in public keys.
 
```

Comment:
This does not seem to say what to do with algorithms without defined seed expansion, or that seed and private key are supposed to match if both are present (AKP is supposed to be much more generic than just ML-DSA). Copying earlier suggested text as suggested change:

```suggestion

If the associated "alg" value does not define seed expansion, then "seed" parameter MUST NOT be present, and the "priv" parameter is used for the private key, if present.

If both "seed" and "priv" parameters are present, then the seed in "seed" parameter MUST expand to the key in "priv" parameter following the seed expansion procedure for the associated "alg" value.
```